### ISSUE TITLE
[Broken Website] Domino's Pizza unable to pay for order

### ISSUE THREAD

**Issue by fredddd1** [darkreader/darkreader#5521]
# Broken Website

## Website
dominos.be

## How to Reproduce

1. Pick your items and try to finalise the order via an online payment method.

## Expected behavior

Being able to pay for the order which works fine with Dark Reader disabled.

## Actual behavior

Picking your payment option will cause for it to be stuck in a loading process for a few minutes, which will eventually time out with an error.

## Screenshots

https://user-images.githubusercontent.com/80743272/114188445-733d6d80-9949-11eb-8450-13da7dfd13e9.mp4

## System Information

- OS: Windows 20H2 64-bit Pro
- Browser:  Google Chrome Version 89.0.4389.114 (Official Build) (64-bit)
- Dark Reader Version:  4.9.31

## Additional context

This issue only exists since a few months ago. At first I thought it was due to too restrictive cookie settings, but only recently I learned Dark Reader was the cause.

**Comment by AndrewDaGuy** [comment]
I'm pretty sure DR only fiddles with the CSS and nothing else, maybe uninstall it and see if the problem persists?

**Comment by fredddd1** [comment]
> I'm pretty sure DR only fiddles with the CSS and nothing else, maybe uninstall it and see if the problem persists?

If I turn Dark Reader on the "Off" status for the website, I can order just fine.

I do not believe Dark Reader messes only with the CSS. Lately I need to turn off Dark Reader for many websites:

Nextcloud Tasks: Unable to load tasks as it will cause the browser to disconnect from the server
Coindera: If Dark Reader is enabled you will be logged out right away after logging in

And now with the Domino's website I realised it was the cause of not being able to place online orders.

**Comment by Gusted** [comment]
> I do not believe Dark Reader messes only with the CSS. Lately I need to turn off Dark Reader for many websites:

Correcto, we insert a small proxy script that modify certain behavior about CSS styles. However I don't see how it could not let you order from domino's. I cannot reproduce this issue as it will require to put a innocent address + some innocent Phone number as there aren't a clear error in the console. I have a hard time thinking how.

Regards,
Gusted

**Comment by fredddd1** [comment]
> > I do not believe Dark Reader messes only with the CSS. Lately I need to turn off Dark Reader for many websites:
> 
> Correcto, we insert a small proxy script that modify certain behavior about CSS styles. However I don't see how it could not let you order from domino's. I cannot reproduce this issue as it will require to put a innocent address + some innocent Phone number as there aren't a clear error in the console. I have a hard time thinking how.
> 
> Regards,
> Gusted

I can private message you my address for you to attempt this yourself. As long as the online payment is not finished, the order is not put through.

**Comment by Gusted** [comment]
> I can private message you my address for you to attempt this yourself. As long as the online payment is not finished, the order is not put through.

Sure thing, otherwise you got a free pizza.

williamzijl7@hotmail.com

**Comment by Gusted** [comment]
Hi!

I had a check with order and at first glance seemed to be reproduce-able, but magically after ~10 seconds it redirected me to the order page. So checking again it seems to be that with dark reader enabled it needed more time to calculate and do the right mechanism to navigate to the bank page.

Regards,
Gusted

**Comment by fredddd1** [comment]
> Hi!
> 
> I had a check with order and at first glance seemed to be reproduce-able, but magically after ~10 seconds it redirected me to the order page. So checking again it seems to be that with dark reader enabled it needed more time to calculate and do the right mechanism to navigate to the bank page.
> 
> Regards,
> Gusted

For me it times out every single time, and I have a pretty fast machine.

What browser are you using?

**Comment by Gusted** [comment]
> What browser are you using?

Firefox v87.
Let me test on chromium. But shouldn't be causing such issue.

**Comment by Gusted** [comment]
> Let me test on chromium. But shouldn't be causing such issue.

Sure thing it was ~5 seconds before having a nice redirect.

**Comment by fredddd1** [comment]
I will close it for now and try to keep testing different scenario's.

**Comment by fredddd1** [comment]
@Gusted I am still trying to fix this, as the issue is still present. The occurrence is very irregular, but when it does happen it can always be fixed by disabling Dark Reader.

Currently I can reproduce it this way:
- VirtualBox with Windows 20H2 64-bit
- Google Chrome
- Dark Reader

Would it be possible for you to attempt this setup?

**Comment by Gusted** [comment]
> Would it be possible for you to attempt this setup?

Yes, I _could_, however to just confirm a theory. Their's currently 1 way for dark reader to break a site and that's when a site use a `document.styleSheets`, an rather obsolete function but somehow still used and it includes our styles and when it wants to do something with it the browser will error it. However while debugging I somehow got it reproduced and tried to fiddle around why, but as for now the code is obfuscated and reading what's doing is hard. So Il give it another try tomorrow as Il better be ending up mailing domino's and hopefully someone from the engineering team can pass me some information what's happening.
